### PR TITLE
make Logger.structuredLogger default for browser

### PR DESCRIPTION
**Pull request by jessekelly881**
Logger.structuredLogger makes a better default for the browser. I wonder if maybe this is too opinionated though.

### CODE REVIEW

**Comment by tim-smart** [comment]
I'm thinking it is a bit too opinionated as it is, as Logger.replace adds the logger regardless (even if the default logger has already been replaced).

I think if we only replace the logger if the default one is present then it might be ok.

**Comment by tim-smart** [comment]
> Oh I wasn't aware of that. Replacing only the default logger and not if it was overridden was my intention. Hmm. What's the correct way to do that then? By adding a `Logger.get` and `Logger.Equivalence` and testing `Logger.get(Logger.defaultLogger)` against `Logger.defaultLogger` to see if it has been overwritten?

Layer.locallyWith that uses a HashSet.has check before replacing the default logger.

**Comment by jessekelly881** [comment]
Something like this? This is the same as doing Logger.replace though, no? I'm probably missing something jaja

```ts
  const setDefaultLogger = Layer.fiberRefLocallyScopedWith(FiberRef.currentLoggers, (loggers) =>  {
    if(HashSet.has(loggers, Logger.defaultLogger)) {
      return loggers;
    }
    const set = HashSet.remove(Logger.defaultLogger);
    return HashSet.add(set as any, Logger.structuredLogger)
  })
```

File: packages/platform-browser/src/internal/runtime.ts [hunk 1]
```diff
@@ -2,14 +2,28 @@ import type { RunMain } from "@effect/platform/Runtime"
...
+
 
 /** @internal */
 export const runMain: RunMain = (
   effect,
   options
 ) => {
-  const _effect = Effect.provide(effect, Logger.structured)
-
+  const _effect = Effect.provide(effect, useStructuredLogger)
```
Comment (tim-smart) [inline packages/platform-browser/src/internal/runtime.ts:25]: I think instead of using a Layer, we should just use `Effect.locallyWith`

Comment (jessekelly881) [reply packages/platform-browser/src/internal/runtime.ts]: Updated

Comment (tim-smart) [reply packages/platform-browser/src/internal/runtime.ts]: I'll hand it over to eslint for the rest of the review :)

**Pull request merged**